code-challenges.test.js

```diff
@@ -14,38 +14,149 @@
+
+    // Create FUNCTION "sentenceGenerator" that takes in ARRAY
+    // Need to RETURN the passed in array and .MAP over each Array(VALUE) to .split the names
+    // So the .map HOF needs to Split on the VALUE.NAME element on each Array element so I can target the 1st letters in the names next <-- <-- (set as VAR "splitArray")
+    // Now take the newly created array of split values (as VAR uppercaseArray) and .toUpperCase the names at the indexes **[0][0]** concatnate with + and .substring to bring letters/words back together. 
+    // Should be able to now RETURN via STRING INTERPOLATION to bring the newly capitalized NAMES into a sentence with corresponding occupation VALUES.
+
+const sentenceGenerator = (array) => {
+  return array.map((value) => {
+    let splitArray = value.name.split(' ')
+    let uppercaseArray = splitArray[0][0].toUpperCase() + splitArray[0].substring(1) + " " + splitArray[1][0].toUpperCase() + splitArray[1].substring(1)
+    return `${uppercaseArray} is ${value.occupation}`
+  })
+}
+
+
+    // JEST TEST PASS
+
+    // PASS./ code - challenges.test.js
+
+    // sentenceGenerator
+
+    //     ✓ takes in an array of objects and returns an array with a sentence about each person with their name capitalized. (1 ms)
+
+
+
 
 
 // --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
-// Expected output: [ 2, 0, -1, 0 ]
-const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
-// Expected output: [ 2, 1, -1 ]
+describe("onlyNumbers", () => {
+  it("takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3", () => {
+    const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
+    // Expected output: [ 2, 0, -1, 0 ]
+    const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
+    // Expected output: [ 2, 1, -1 ]
+    expect(onlyNumbers(hodgepodge1)).toEqual([2, 0, -1, 0])
+    expect(onlyNumbers(hodgepodge2)).toEqual([2, 1, -1])
+  })
+})
+
+    // JEST "GOOD" FAILURE
+
+    //  ● onlyNumbers › takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3
+
+    // ReferenceError: onlyNumbers is not defined
 
 
 // b) Create the function that makes the test pass.
 
+    // Fuction ONLYNUMBERS takes in ARRAY
+    // FILTER over the ARRAY elements -> Condition? TYPEOF === NUMBER     <<<< set as variable getNumbers
+    // MAP over newArray elements and %= 3 on each     <<<< set as variable remainders
+    // RETURN remainders
+
+const onlyNumbers = (array) => {
+  let getNumbers = array.filter(element => typeof element === 'number')
+  let remainders = getNumbers.map(element => element %= 3)
+  return remainders
+}
```

Comment:
Awesome!  You can condense this down by chaining the map after the filter and returning that same line instead of assigning to a variable.  Just a fun stretch refactor if you wish to try it.